pysnn/neuron.py

```diff
@@ -137,8 +137,11 @@ def refrac(self, spikes):
 
         Can be overwritten in case of the need of more refined functionality.
         """
-        self.refrac_counts[self.refrac_counts > 0] -= self.dt
-        self.refrac_counts += self.duration_refrac * self.convert_spikes(spikes)
+        if self.duration_refrac > 1:
+            self.refrac_counts[self.refrac_counts > 0] -= self.dt
```

Comment:
Looking at this, I see I initially never thought about the situation where the refrac count drops below 0. With this change pending, might also want to add lower bound clamping at 0.

Reply:
I think an assert for `duration_refrac >= 0` next to the `duration_refrac % dt == 0` during initialization will solve this and prevent it from ever going below 0